packages/grid/x-data-grid-pro/src/tests/detailPanel.DataGridPro.test.tsx

```diff
@@ -101,10 +100,9 @@ describe('<DataGridPro /> - Detail panel', () => {
       />,
     );
     fireEvent.click(screen.getAllByRole('button', { name: 'Expand' })[0]);
+    await act(() => Promise.resolve());
```

Comment:
Could this change have been the culprit of now quite frequent false-positive [failures in PR CI](https://app.circleci.com/pipelines/github/mui/mui-x/34421/workflows/e839d032-6bc0-4de1-b0a3-b25b08a2056b/jobs/197440)? 🤔

Reply:
Yeah, the tests were failing with `waitFor`.
I started digging into it and realized that `waitFor` only calls the callback once and then it fails because of the timeout.
Then it turned out that `waitFor` cannot work correctly with `fake` clock, because it uses `setTimeout` under the hood 🙂

I think we should get rid of the fake clock and use it only when needed. This would allow using `waitFor` instead of a hacks like `clock.tick()` and `await act(() => Promise.resolve()); `

Reply:
https://github.com/mui/mui-x/pull/8097

Reply:
> I think we should get rid of the fake clock and use it only when needed. This would allow using `waitFor` instead of a hacks like `clock.tick()` and `await act(() => Promise.resolve()); `

Oh yes, those are really great points you noticed. 👍 
I'm all for only going with `fake` timers when it case calls for it, but other than that—our tests should try mimicking real use cases as possible, that is why `testing-library` has such utilities. 🙈